packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/DiagramRenderer.tsx

```diff
@@ -111,7 +116,25 @@ export const DiagramRenderer = ({
   const { updateSelectionOnNodesChange, updateSelectionOnEdgesChange } = useDiagramSelection(selection, setSelection);
 
   const handleNodesChange: OnNodesChange = (changes: NodeChange[]) => {
-    onNodesChange(onBorderChange(onHandleChange(changes)));
+    const nodeChanges = onBorderChange(onHandleChange(changes));
+    const firstNodeChange = nodeChanges[0];
+    if (firstNodeChange) {
+      if (firstNodeChange.type === 'position') {
+        if (typeof firstNodeChange.dragging === 'boolean' && !firstNodeChange.dragging) {
+          if (!hasDroppedNodeParentChanged()) {
+            const diagramToLayout: RawDiagram = {
+              nodes: nodes as Node<NodeData, DiagramNodeType>[],
+              edges,
+            };
+            layout(diagramToLayout, diagramToLayout, (laidOutDiagram) => {
+              setNodes(laidOutDiagram.nodes);
+              setEdges(laidOutDiagram.edges);
+            });
+          }
+        }
+      }
+    }
```

Comment:
- All this does not belong here, those are details specific to the layout. Have a look at `useBorderChange` to 
encapsulate this kind of logics
- Change the test of  `firstNodeChange.type === 'position'` to cast the change to a `NodePositionChange` instead of testing with `typeof`

Reply:
- I will move the code somewhere else.
- typescript already infers that after the test of `firstNodeChange.type === 'position'` it is a `NodePositionChange`. But It needs to test `dragging` with `typeof` because `dragging` is `boolean | undefined`. So `typeof firstNodeChange.dragging === 'boolean'` is used to ensure `dragging` is present. Reactflow uses `typeof currentChange.dragging !== 'undefined'` but I used what Reactflow does for `nodeChange.resizing` (see https://github.com/wbkd/react-flow/blob/0892e2ea9aaa19a563858b849f6065f966dbf546/packages/core/src/utils/changes.ts)